### PR TITLE
Clean up boilerplate and consolidate documentation

### PR DESCRIPTION
**Pull request by palmskog**
I deleted the old boilerplate mentioned in #47 and modernized the makefile. I also consolidated the existing documentation to `meta.yml`.

### CODE REVIEW

**Comment by palmskog** [comment]
@proux01 anything I need to do to progress on this PR? I tried to make it uncontroversial - no documentation is removed, just moved to meta/README where it can be found by more people.

File: meta.yml [hunk 1]
```diff
@@ -123,9 +124,74 @@ categories:
...
+  ## Refinements
+
+  The refinements directory has the following content:
+
+  - `refinements`: Classes for refinements and refines together with
+    operational typeclasses for common operations.
+
+  - `binnat`: Proof that the binary naturals of Coq (`N`) is a refinement
+    of SSReflect unary naturals (`nat`) together with basic operations.
+
```
Comment (proux01) [inline meta.yml:157]: Here we could add
```suggestion
  - `binnat`: Proof that the binary naturals of Coq (`N`) is a refinement
    of SSReflect unary naturals (`nat`) together with basic operations.

- `binord`: Proof that the binary natural numbers of Coq (`N`) is a refinement
  of SSReflect ordinals.
```

File: meta.yml [hunk 2]
```diff
@@ -123,9 +124,74 @@ categories:
...
+  - `binint`: SSReflect integers (`ssrint`) are refined to a new type
+    paremetrized by positive numbers (represented by a sigma type) and
+    natural numbers.  This means that proofs can be done using only
+    lemmas from the SSReflect library which leads to simpler proofs than
+    previous versions of `binint` (e.g., `N`).
+
```
Comment (proux01) [inline meta.yml:163]: Here we could add
```suggestion

- `binrat`: arbitrary precision rational numbers (`bigQ`) from the [bignums](https://github.com/coq/bignums)
  library are refined to MathComp's rationals (`rat`).
```

File: meta.yml [hunk 3]
```diff
@@ -123,9 +124,74 @@ categories:
...
+  The refinements directory has the following content:
+
+  - `refinements`: Classes for refinements and refines together with
+    operational typeclasses for common operations.
+
+  - `binnat`: Proof that the binary naturals of Coq (`N`) is a refinement
+    of SSReflect unary naturals (`nat`) together with basic operations.
+
+  - `binint`: SSReflect integers (`ssrint`) are refined to a new type
```
Comment (proux01) [inline meta.yml:158]:
```suggestion
  - `binint`: MathComp integers (`ssrint`) are refined to a new type
```

File: meta.yml [hunk 4]
```diff
@@ -123,9 +124,74 @@ categories:
...
+    lemmas from the SSReflect library which leads to simpler proofs than
+    previous versions of `binint` (e.g., `N`).
+
+  - `rational`: The rational numbers of SSReflect (`rat`) is refined to
+    pairs of elements refining integers using parametricity of
+    refinements.
+
+  - `seqmatrix`: First and incomplete attempt to refine SSReflect
+    matrices `M[R]_(m,n)`) to lists of lists (`seq (seq R)`).
+    Work in progress.
```
Comment (proux01) [inline meta.yml:170]:
```suggestion
  - `seqmatrix` and `seqmx_complements`: refinement of MathComp
    matrices `M[R]_(m,n)`) to lists of lists (`seq (seq R)`).
```

File: meta.yml [hunk 5]
```diff
@@ -123,9 +124,74 @@ categories:
...
   
-  - releases: archives of pre-GitHub releases
+  - `seqpoly`: First and incomplete attempt to refine SSReflect
+    polynomials (`{poly R}`) to lists (`seq R`). Work in progress.
```
Comment (proux01) [inline meta.yml:173]:
```suggestion
  - `seqpoly`: refinement of MathComp
    polynomials (`{poly R}`) to lists (`seq R`).
```

File: meta.yml [hunk 6]
```diff
@@ -123,9 +124,74 @@ categories:
...
+
```
Comment (proux01) [inline meta.yml:174]:
```suggestion

- `multipoly`: refinement of [MathComp multinomials](https://github.com/math-comp/multinomials/)
  multivariate polynomials by [standard library maps](https://github.com/coq/coq/blob/master/theories/FSets/FMapAVL.v).
```

**Review by proux01 — APPROVED** [review]
Sorry, looks like I forgot about that one, thanks for the reminder.

**Comment by proux01** [comment]
Thanks @palmskog !

**Pull request merged**